packages/url-loader/src/plugins/abr.ts

```diff
@@ -14,14 +14,11 @@ export const AbrPlugin = plugin({
   name: "Abr",
   supports: "video",
   inferOwnOptions: {} as AbrPlugin.Options,
-  applyWhen: 'streamingProfile',
   props: {
     streamingProfile: true,
   },
   apply: (asset, opts) => {
-    if (typeof opts.streamingProfile === "string") {
```

Comment:
so my only concern here is that this is explicitly checking for a string, but it doesnt seem that the other side is checking the data type, so someone could pass in any truthy value and it would pass it along to the transformation?

Reply:
This kind of falls under the previous conversation we had about runtime validation. Since they are providing typed input (the `streamingProfile` prop already enforces that the value is a string), passing `true` or something should be a type error.

Reply:
would that be a type error if they're not using Typescript?

Reply:
You can enable `checkJs` even if they're writing JS.

Based on our previous conversation, my impression was that we're assuming that people are using the API according to the types or the docs, i.e. not passing garbage. I think this is a good assumption when building a typed API, as otherwise you add tons of bloat for error handling that should never really occur based on the contract in general.

If there was a reason to expect untyped input/lots of mistakes, we'd kinda be back to square one wanting a runtime validator like Zod :P

Reply:
is checkJs a user setting or library setting? guessing user setting? so idk if thats a good assumption

i think its more likely people will be using typescript in a Next.js app these days given its a default but i think there still need to be handling, but i guess my impression of the conversation wasn't whether or not to include the runtime validation i already had, moreso not using Zod for it, which i think was only being used in one or two places

seems like its just regressing things a hair for negligible value since it doesnt rely on zod? most of the checks that were there were simpler checks and most relied on specific code paths as opposed to simple validation

Reply:
i agree that its not my responsibility but i do believe if i can provide guide rails that will have negligible impact on the code/project but have a beneficial impact on their developer experience, i think its worth the effort, like this isn't really going to break the bank in terms of performance but it can help avoid broken experiences

what will happen is instead of getting a broken video because of an improper value (like a number), which can be very difficult to debug through Cloudinary (like not everyone knows about x-cld-error header), it will not add this parameter to the URL maintaining the video still in a working experience

an enhancement would be to warn of the usage in development mode like i do in the `if (!getFormat(defaultImage)) {` below but i opted not to specifically do that in every instance

Reply:
Okay I will add those checks back in!